domains/measurements/Measurements.API/API/Measurements/gRPC/V1/Services/MeasurementsService.cs

```diff
@@ -0,0 +1,75 @@
+    {
+        _client = client;
+    }
+
+    public override async Task<GetMeasurementsResponse> GetMeasurements(GetMeasurementsRequest request, ServerCallContext context)
+    {
+        var dhRequest = new MeterTimeSeriesRequest
+        {
+            Actor = request.Actor,
+            Subject = request.Subject,
+            Gsrn = request.Gsrn,
+            DateFrom = request.DateFrom,
+            DateTo = request.DateTo
+        };
+
+        var dhResponse = await _client.GetMeterTimeSeriesAsync(dhRequest);
+
+        if (dhResponse.GetMeterTimeSeriesRejection != null && dhResponse.GetMeterTimeSeriesRejection.Rejection.Count > 0)
+        {
+            var rejections = string.Join(
+                "\n",
+                dhResponse.GetMeterTimeSeriesRejection.Rejection
+                    .Select(rejection => $"MeteringPoint {rejection.MeteringPointId}\n ErrorCode {rejection.ErrorCode}"));
+
+            throw new DataHubFacadeException($"GetMeterTimeSeries was rejected\n {rejections}");
+        }
+
+        var measurements = dhResponse.GetMeterTimeSeriesResult.MeterTimeSeriesMeteringPoint.SelectMany(
+            mp => mp.MeteringPointStates?.SelectMany(
+                state => state.NonProfiledEnergyQuantities?.SelectMany(
+                    item => item.EnergyQuantityValues?.Select(
+                        quantity => new Measurement
+                        {
+                            Gsrn = mp.MeteringPointId,
+                            DateFrom = TimeSeriesHelper.GetDateTimeFromMeterReadingOccurrence(item.Date, int.Parse(quantity.Position) - 1, state.MeterReadingOccurrence),
+                            DateTo = TimeSeriesHelper.GetDateTimeFromMeterReadingOccurrence(item.Date, int.Parse(quantity.Position), state.MeterReadingOccurrence),
+                            Quantity = TimeSeriesHelper.GetQuantityFromMeterReading(quantity.EnergyTimeSeriesMeasureUnit, quantity.EnergyQuantity),
+                            Quality = TimeSeriesHelper.GetQuantityQualityFromMeterReading(quantity.QuantityQuality),
+                        }
+                    ) ?? Enumerable.Empty<Measurement>()
+                )
+                .GroupBy(measurement => measurement.DateFrom.ZeroedHour()) 
+                    .Select(group => new Measurement
+                    {
+                        Gsrn = mp.MeteringPointId,
+                        DateFrom = group.Min(m => m.DateFrom),
+                        DateTo = group.Max(m => m.DateTo),
+                        Quantity = group.Sum(m => m.Quantity),
+                        Quality = group.Max(m => m.Quality),
+                    })
+                .Where(measurement => measurement.DateFrom >= request.DateFrom && measurement.DateTo <= request.DateTo)
+                 ?? Enumerable.Empty<Measurement>()
+            ) ?? Enumerable.Empty<Measurement>()
+        ) ?? Enumerable.Empty<Measurement>();
```

Comment:
My IDE tells me line`69` is redundant.

Reply:
That is under the assumption that you can't set gRPC messages to null. I'm note sure - I've seen it done in an integration test, so I'll leave it just in case